pkg/cmd/cli/cmd/deploy.go

```diff
@@ -222,7 +222,7 @@ func (o DeployOptions) deploy(config *deployapi.DeploymentConfig, out io.Writer)
 	}
 
 	config.Status.LatestVersion++
-	_, err = o.osClient.DeploymentConfigs(config.Namespace).Update(config)
+	_, err = o.osClient.DeploymentConfigs(config.Namespace).UpdateStatus(config)
```

Comment:
FWIW in the short term, updates to the status won't result in a new deployment unless `status.latestVersion` changes.